pkg/operator/controller/ingress/controller.go

```diff
@@ -309,6 +314,17 @@ func (r *reconciler) admit(current *operatorv1.IngressController, ingressConfig
 		Reason: "Valid",
 	})
 	updated.Status.ObservedGeneration = updated.Generation
+
+	if !manageDNSForDomain(updated.Status.Domain, platformStatus, dnsConfig) {
+		updated.Status.Conditions = MergeConditions(updated.Status.Conditions, operatorv1.OperatorCondition{
+			Type:    operatorv1.DNSManagedIngressConditionType,
+			Status:  operatorv1.ConditionFalse,
+			Reason:  "DomainNotMatching",
+			Message: "DNS management is not supported for ingresscontrollers with domains not matching the baseDomain of the cluster DNS config.",
+		})
```

Comment:
Do you need to set the condition in both `admit` and `computeDNSStatus`?  After admission, the reconcile loop should queue another reconciliation, and then `syncIngressControllerStatus` should update the status conditions, including the "DNSManaged" status condition.  It makes sense to me that `admit` only set the "Admitted" status condition, and possibly emit admission-related events, and `syncIngressControllerStatus` set other status conditions.

Reply:
I was also not sure about this. I'll then delete the one in controller.go and only keep the one in status.go.